src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -8,111 +8,63 @@
         {
-            SslContext sslContext = new SslContext
-            {
-                isServer = isServer,
-            };
+            SafeSslHandle context = null;
 
-            try
-            {
-                IntPtr method = GetSslMethod(isServer, options);
-
-                IntPtr contextPtr = libssl.SSL_CTX_new(method);
+            IntPtr method = GetSslMethod(isServer, options);
 
-                if (IntPtr.Zero == contextPtr)
+            using (libssl.SafeSslContextHandle innerContext = new libssl.SafeSslContextHandle(method))
+            {
+                if (innerContext.IsInvalid)
                 {
                     throw CreateSslException("Failed to allocate SSL/TLS context");
                 }
 
-                libssl.SSL_CTX_ctrl(contextPtr, libssl.SSL_CTRL_OPTIONS, options, IntPtr.Zero);
+                libssl.SSL_CTX_ctrl(innerContext, libssl.SSL_CTRL_OPTIONS, options, IntPtr.Zero);
 
-                libssl.SSL_CTX_set_quiet_shutdown(contextPtr, 1);
+                libssl.SSL_CTX_set_quiet_shutdown(innerContext, 1);
 
                 if (certHandle != null && certKeyHandle != null)
                 {
-                    SetSslCertificate(contextPtr, certHandle, certKeyHandle);
-                }
-
-                sslContext.sslPtr = libssl.SSL_new(contextPtr);
-
-                libssl.SSL_CTX_free(contextPtr);
-
-                if (IntPtr.Zero == sslContext.sslPtr)
-                {
-                    throw CreateSslException("Failed to create SSSL object from SSL context");
-                }
-
-                IntPtr memMethod = libcrypto.BIO_s_mem();
-
-                if (IntPtr.Zero == memMethod)
-                {
-                    throw CreateSslException("Failed to return memory BIO method function");
+                    SetSslCertificate(innerContext, certHandle, certKeyHandle);
                 }
 
-                sslContext.readBioPtr = libssl.BIO_new(memMethod);
-                sslContext.writeBioPtr = libssl.BIO_new(memMethod);
-
-                if ((IntPtr.Zero == sslContext.readBioPtr) || (IntPtr.Zero == sslContext.writeBioPtr))
-                {
-                    FreeBio(sslContext);
-                    throw CreateSslException("Failed to retun new BIO for a given method type");
-                }
-
-                if (isServer)
-                {
-                    libssl.SSL_set_accept_state(sslContext.sslPtr);
-                }
-                else
-                {
-                    libssl.SSL_set_connect_state(sslContext.sslPtr);
-                }
-
-                libssl.SSL_set_bio(sslContext.sslPtr, sslContext.readBioPtr, sslContext.writeBioPtr);
+                context = SafeSslHandle.Create(innerContext, isServer);
             }
-            catch
+
+            if ((null != context) && !context.IsInvalid)
```

Comment:
Sorry.. it was due to an earlier avatar of Create which was returning null. Fixed this now